### DISCUSSION TITLE
Name Squatting

### DISCUSSION THREAD

**Discussion by itrTie** [community/community#28516] [Other Feature Feedback, Questions, & Ideas]
I found an account that has been inactive for ~9 months, and I would like the username.  Despite the policy outlined [here](https://docs.github.com/en/site-policy/other-site-policies/github-username-policy), I can't send a support ticket or report the user.

If I can report it, why are there no methods of reporting, it?
If I can't report it, why do the docs say that I can?

GitHub docs say:
> GitHub prohibits account name squatting, and account names may not be reserved or **inactively held for future use**. Accounts violating this name squatting policy may be removed or renamed without notice. Attempts to sell, buy, or solicit other forms of payment in exchange for account names are prohibited and may result in permanent account suspension.

No other discussions have given a clear answer or have linked to a github.community discussion which only redirects to this home page.

**Comment by radar-blip** [comment]
If that username has only been inactive for 9 months, I would forget about it, pick another one. There are a lot of reasons a person could be away for 9 months.

**Reply by itrTie** [reply]
The account was created in September 2021, and there was no activity since then.  I can see no activity for an extended period, but this was no activity ever.

**Reply by radar-blip** [reply]
I dont know what to tell you. I dont know how long they let usernames sit dormant. 
By the way, the link you put up, isn't working (not for me anyway).
Any other username wont do?

**Comment by billythekid** [comment]
Howdy @itrTie! 👋 

There have been several posts like this on the forum over time, although I'm not sure if I could find them all after the recent migration to Discussions so hopefully this reply can clear this up for you once and for all.

This is the doc to which you refer: [GitHub Username Policy](https://docs.github.com/site-policy/other-site-policies/github-username-policy).

The docs are correct. Name squatting is not allowed. That means people should not create accounts to hold onto the username. They should definitely not try to buy or sell a username. 🙅 

It _doesn't_ mean we'll accept requests from people asking for a username that's been taken. They're given out on a first-come first-served basis. That same doc tells us what to do if the username we want is unavailable already in the first section `What if the username I want is already taken?`:

> If the username you want has already been claimed, consider other names or unique variations. Using a number, hyphen, or an alternative spelling might help you identify a desirable username still available.

The username change tool will let you know if a username is available. (Not all accounts are publicly visible on GitHub and not all endpoints are available for registration as accounts. Unfortunately usernames share the same namespace as everything else. We can't put usernames behind a /u/ path or similar nowadays because _everything will break_! 😬)

GitHub _may_ take action on an account thought to be name squatting but even then...

> Accounts violating this name squatting policy may be removed or renamed without notice.

...that doesn't say the username will be made available to be used by someone else.

It does mention in that document but I think it's worth restating here that it's important to note that lots of GitHub accounts are being actively used in private. Not all activity on GitHub is public and some activity doesn't even require a user to log in to the github.com website. Just because an account looks abandoned to the public view, doesn't mean it is.

So in summary;

1. It's NOT allowed to register an account to lie unused, so please don't do this.
2. We don't, at this time, generally accept requests for taken usernames. Please choose a different username or a variation if you are unable to select the username you try at first.

This doesn't mean that we don't enforce the policy. A lot of work goes on behind the scenes in this area to keep the platform safe and usable for everyone.

If you find that your account has been suspended or hidden and you think it might have happened because you've been found to be name squatting or inactively holding an account, please open a [reinstatement request](https://support.github.com/contact/reinstatement) where someone will let you know what's happened and what you need to do, if anything, to remedy the situation.

For more information, read [GitHub Appeal and Reinstatement](https://docs.github.com/site-policy/acceptable-use-policies/github-appeal-and-reinstatement) in the GitHub Docs.

I appreciate there is some nuance here which may be different from how the policy is often interpreted by those who wish to _claim_ a username for their own use, so I hope that clears this up.

Finally, the [Changing your GitHub username](https://docs.github.com/github/setting-up-and-managing-your-github-user-account/changing-your-github-username) document lets you know some of the impact this can have on an account. (Changing the case of your username is fine, paths here are case insensitive for these purposes!) I would not recommend anyone changes their username, especially if they have an established account or have published code to package managers etc. It _can_ become a bit of a headache so tread carefully!

Cheers,
Billy 🤠

**Reply by itrTie** [reply]
If I'm understanding correctly, name squatting isn't allowed but you don't want people to report these cases.  Seems a bit counterintuitive, but I guess there is some sort of reason.

Thanks!

**Reply by o6uoq** [reply]
They don't want the additional support requests. Terrible contradiction of policy to claim one thing has merit, but no avenue for customers' to pursue - as per their OWN guidelines!

**Reply by henrijd** [reply]
@billythekid are atleast name squatting requests of pro users accepted?

**Reply by nadilas** [reply]
It doesn't seem like it

<img width="776" alt="image" src="https://user-images.githubusercontent.com/5324856/188271345-0f16dd68-ac0d-4e00-99cb-28008f01e19b.png">

**Reply by henrijd** [reply]
well it isn't explicit that both support types community or standard don't accept the requests, maybe pro users can make a request like in GitLab.

![brave_SMyi5WKILL](https://user-images.githubusercontent.com/36113207/188278736-057888fd-b6ad-4b08-99b6-90fcc5087d4d.png)

**Reply by nadilas** [reply]
@henrijd entirely possible. I checked from an enterprise account though

**Reply by stretch07** [reply]
this is so annoying

**Reply by billythekid** [reply]
Sorry for the delayed response here, @henrijd, the subscription the requester is on makes no difference. 

Paid plans just mean you get access to the support team directly via tickets in the first place, but that doesn't mean what we do is different with regards to how we approach the issue. Free plans are community support (this discussions area) only, which is why I've put an answer here so all users can see what the state of play is with this.

See https://github.com/pricing#compare-features (consider Team (orgs) equivalent to Pro (users) in this regards)

![support-plans](https://user-images.githubusercontent.com/330170/190599937-79b4add6-ea6a-4de1-bf23-c383ecbce838.png)

While you could go to the effort of creating a ticket as an account holder on a paid plan, it's likely the answer you'd receive would be disappointing, the only difference being you'd be hearing it from a member of staff rather than a community volunteer.

**Reply by henrijd** [reply]
@billythekid This is bad if even paying clients can't change it.

I understand that GitHub got a lot of name squatting issue requests back then and resolved to stop it completely since it would be costing many of its support resources for basically no reward and costing time for other more important issues. But stopping it for even paying clients, is really an unneccesary punishment for both the clients and GitHub itself, since they'd be losing possible revenue from users willing to pay for a subscription just to change their username and maybe even stay if they so like. Having alteast the Option would be a big win instead of being completely cut off of the possibility. 

So please if possible forward my suggestion for them to reconsider this situation for a brief action on this matter.

**Comment by o6uoq** [comment]
If someone on a paid plan opens a support ticket, it would be great to know if it is a canned response or paid = actually something done.

**Reply by stretch07** [reply]
It feels unfair that paid clients can resolve their name squatting issues but free clients can’t
Kinda as if you have to pay to get the username you want on github

**Reply by billythekid** [reply]
Hi folks, I talk a bit about paid vs free here:

https://github.com/community/community/discussions/28516#discussioncomment-3660965

TLDR; it's the same outcome.

Cheers,
Billy 🤠
